src/diffusers/models/modeling_utils.py

```diff
@@ -579,10 +579,17 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                             " those weights or else make sure your checkpoint file is correct."
                         )
 
+                    empty_state_dict = model.state_dict()
```

Comment:
Is it worthwhile to `del` this when we are done with it?

Reply:
It's a meta device dict so doesn't hold any memory. So don't think it's worth it :-)

Reply:
Ah, ok, awesome!

Reply:
What is a meta device here? Asking for my curiosity.

Reply:
Essentially doing:
```py
import torch

ten = torch.ones((10**15,), device="meta")
```

is instant as it doesn't allocate any memory. Would not recommend removing the `device="meta"` there though ;-) 

We use this to not "double" allocate memory when loading pretrained weights.